Utils/Campaign builder/Unit1.pas

```diff
@@ -277,15 +309,25 @@ procedure TForm1.edtShortNameChange(Sender: TObject);
 var
   cmp: TKMCampaignId;
 begin
-  if fUpdating then Exit;
+  if Length(edtShortName.Text) = 3 then
+  begin
+    if fUpdating then Exit;
 
-  cmp[0] := Ord(edtShortName.Text[1]);
-  cmp[1] := Ord(edtShortName.Text[2]);
-  cmp[2] := Ord(edtShortName.Text[3]);
-  C.CampaignId := cmp;
+    cmp[0] := Ord(edtShortName.Text[1]);
+    cmp[1] := Ord(edtShortName.Text[2]);
+    cmp[2] := Ord(edtShortName.Text[3]);
+    C.CampaignId := cmp;
 
-  //Shortname may be used as mapname in List
-  UpdateList;
+    //Shortname may be used as mapname in List
+    UpdateList;
+  end;
+end;
+
+//This method corrects the issue so that the first character has to be
+//administered twice when entering the reduced campaign name
+procedure TForm1.edtShortNameClick(Sender: TObject);
+begin
+  edtShortName.Text := edtShortName.Text;

```

Comment:
There should be a better way to fix it than this. If you need to trigger the OnChange event you can run it manually